src/backend/dx12/src/command.rs

```diff
@@ -705,8 +705,19 @@ impl com::RawCommandBuffer<Backend> for CommandBuffer {
 
         // transition barriers
         for barrier in barriers {
-            let barrier = barrier.borrow();
-            match *barrier {
+            match *barrier.borrow() {
+                memory::Barrier::AllBuffers(_) |
+                memory::Barrier::AllImages(_) => {
+                    let mut bar = d3d12::D3D12_RESOURCE_BARRIER {
+                        Type: d3d12::D3D12_RESOURCE_BARRIER_TYPE_UAV,
+                        Flags: d3d12::D3D12_RESOURCE_BARRIER_FLAG_NONE,
+                        u: unsafe { mem::zeroed() },
+                    };
+                    *unsafe { bar.u.UAV_mut() } = d3d12::D3D12_RESOURCE_UAV_BARRIER {
```

Comment:
So we have confirmation that it's fine with before-after state mismatches?

Reply:
Could you also add a comment here that UAV barriers provide the necessary sync behaviour?